HelloWorldFunction/src/main/java/helloworld/App.java

```diff
@@ -22,102 +22,109 @@
+            String ddbtestget = this.GetItem("HelloWorldTable", "id", id.toString());
 
-            String output = String.format("{ \"message\": \"hello world\", \"location\": \"%s\" \n \"put\":\"%s\"-\"%s\" \n \"get\":\"%s\"-\"%s\"}", pageContents,id.toString(),ddbtestput,id.toString(),ddbtestget);
+            String output = String.format(
+                    "{ \"message\": \"hello world\", \"location\": \"%s\" \n \"put\":\"%s\"-\"%s\" \n \"get\":\"%s\"-\"%s\"}",
+                    pageContents, id.toString(), ddbtestput, id.toString(), ddbtestget);
             return new GatewayResponse(output, headers, 200);
         } catch (IOException e) {
             return new GatewayResponse("{}", headers, 500);
         }
     }
 
-    private String getPageContents(String address) throws IOException{
+    private String getPageContents(String address) throws IOException {
         URL url = new URL(address);
-        try(BufferedReader br = new BufferedReader(new InputStreamReader(url.openStream()))) {
+        try (BufferedReader br = new BufferedReader(new InputStreamReader(url.openStream()))) {
             return br.lines().collect(Collectors.joining(System.lineSeparator()));
         }
     }
 
-    private String GetItem(String TableName, String PrimaryKey, String Value) //throws AmazonServiceException
+    private String GetItem(String TableName, String PrimaryKey, String Value) // throws
+                                                                                                // AmazonServiceException
     {
 
-        HashMap<String,AttributeValue> key_to_get =
-        new HashMap<String,AttributeValue>();
-    
+        HashMap<String, AttributeValue> key_to_get = new HashMap<String, AttributeValue>();
+
         key_to_get.put(PrimaryKey, new AttributeValue(Value));
-        
+
         GetItemRequest request = null;
 
-        request = new GetItemRequest()
-            .withKey(key_to_get)
-            .withTableName(TableName);
-        
-        final AmazonDynamoDB ddb = AmazonDynamoDBClientBuilder.defaultClient();
-        
+        request = new GetItemRequest().withKey(key_to_get).withTableName(TableName);
+
+        AmazonDynamoDB ddb = AmazonDynamoDBClientBuilder.defaultClient();
+
         try {
-            Map<String,AttributeValue> returned_item =
-            ddb.getItem(request).getItem();
+            Map<String, AttributeValue> returned_item = ddb.getItem(request).getItem();
             if (returned_item != null) {
                 Set<String> keys = returned_item.keySet();
-                String output;
-                for (String key : keys) {
-                    System.out.format("%s: %s\n",
-                            key, returned_item.get(key).toString());
+                //String output;
+                for ( String key : keys) {
+                    System.out.format("%s: %s\n", key, returned_item.get(key).toString());
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: Iterating over [map.keySet()](https://docs.oracle.com/javase/8/docs/api/java/util/Map.html#keySet--) and calling [map.get(Key)](https://docs.oracle.com/javase/8/docs/api/java/util/Map.html#get-java.lang.Object-) to get the values within the loop is inefficient. 
**Fix**: Iterate over [map.entrySet()](https://docs.oracle.com/javase/8/docs/api/java/util/Map.html#entrySet--) and access the values using [entry.getValue()](https://docs.oracle.com/javase/8/docs/api/java/util/Map.Entry.html#getValue--) to avoid potential performance penalty due to lookup.